resources/prosody-plugins/token/util.lib.lua

```diff
@@ -177,6 +185,39 @@ local function _verify_token(session, token)
     end
 end
 
+local function _verify_room_and_domain(room_address_to_verify, room, domain)
+
+    local room_to_verify,_,_ = jid.split(room_address_to_verify);
+    if not enableDomainVerification then
+        -- extract room name using all chars, except the not allowed ones
+
+        if room_to_verify == nil then
+            log("error",
+                "Unable to get name of the MUC room ? to: %s", room_address_to_verify);
+            return false;
+        end
+        return room_to_verify == string.lower(room);
+    end
+
+    -- make sure we remove any resource when comparing
+    room_address_to_verify = jid.bare(room_address_to_verify);
+    local target_subdomain, target_room = room_to_verify:match("^%[([^%]]+)%](.+)$");
+    if target_subdomain then
+
+        -- from this point we depend on muc_domain_base, deny access if option is missing
+        if not muc_domain_base then
+            module:log("warn", "No 'muc_domain_base' option set, denying access!");
+            return false;
+        end
+
+        return room_address_to_verify == jid.join("["..domain.."]"..string.lower(room), muc_domain);
+    else
+        -- we do not have a domain part (multidomain is not enabled)
```

Comment:
is there any doc available anywhere that explain this options with some examples ? it seems like there are bunch of possible scenarios, but I'm not able to verify this code without any docs